### ISSUE TITLE
Test for the example in CI

### ISSUE THREAD

**Issue by tom-tan** [common-workflow-language/user_guide#37]
It would be nice if there is a CI tester (such as Travis CI) for the examples in the user guide.
Such a scheme can keep the examples runnable for each commit.

**Comment by tobyhodges** [comment]
This *would* be great! Perhaps I can have a go at configuring such a thing...

**Comment by tom-tan** [comment]
How about using `cwltest` for this purpose?
It makes easier to maintain by using the existing framework.

To use `cwltest` for testing user guide, there are several issues to be solved:
- In user guide, there are several examples that are expected to be failed (e.g. `record.cwl` and `record-job1.yml` in Section 11).
   It would be nice if `cwltest` can handle such cases.
- Some examples just output the string as a side effect and return no results (e.g. `record.cwl` and `record-job2.yml` in Section 11). Also, some examples depend on the results of other examples in previous sections (e.g. `tar.cwl` in Section 4). 
   It would be better to fix the examples to be testable, that is, fix them not to use side effects and to be independent each other.

If it is OK for this plan, I will make issues and pull requests for it later.

**Comment by tobyhodges** [comment]
this sounds like a pretty good plan to me, though I do think that it would be good to keep examples where errors are expected. These are often good for people searching the web for an error message, and/or to demonstrate certain features e.g. dependent arguments/options. 

what do you think, @mr-c?

**Comment by mr-c** [comment]
Yes, we need to be able to have "known failure" testing, so a PR to adding that to `cwltest` would be very welcome!

**Comment by tom-tan** [comment]
Thank you for the reply.

Here is a todo list for this issue.
- [ ] [cwltest] add support for "known failure" tests
- [ ] [user_guide] make the tests in each section independent
- [ ] [user_guide] make the tests not to use side effects

I will make PRs for them.
